pyne/vta.py

```diff
@@ -0,0 +1,481 @@
+    v_1d : float
+        Direction value.
+    bound : list of float
+        Boundary values of current direction.
+
+    Return:
+    -------
+    n_grid : float
+        The coordinate value of next mesh grid.
+    """
+    n_grid = 0.0
+    flag = False
+    if v_1d > 0:
+        for bnd_pos in bound:
+            if cor < bnd_pos:
+                n_grid = bnd_pos
+                flag = True
+                break
+    else:
+        for i in range(len(bound)-1, -1, -1):
+            if cor > bound[i]:
+                n_grid = bound[i]
+                flag = True
+                break
+    if flag:
+        return n_grid
+    else:
+        # point goes out of the mesh
+        return v_1d * float('inf')
+
+
+def _cal_max_travel_length_in_current_voxel(tp, end, v, bounds):
+    """
+    Calculate the maximum travel length in current voxel before reach next
+    boundary.
+    
+    Parameters:
+    -----------
+    tp : numpy array
+        An array of size 3. Temporary point position.
+    end : numpy array
+        An array of size 3. End point
+    v : numpy array
+        Direction vector from start to end: [v_x, v_y, v_z].
+    bounds : list
+        Boundaries of the mesh grid.
+        Expected format: [[x_min, ..., x_max],
+                          [y_min, ..., y_max],
+                          [z_min, ..., z_max]]
+
+    Return:
+    -------
+    t_maxs : numpy array
+        The maximum travel lenght in current voxel for 3 directions. Format: 
+        [t_max_x, t_max_y, t_max_z]. These value could be both positive and
+        negtive. And could be 'inf' or '-inf'.
+    """
+    n_x_grid = _find_next_grid_1d(tp[0], v[0], bounds[0])
+    n_y_grid = _find_next_grid_1d(tp[1], v[1], bounds[1])
+    n_z_grid = _find_next_grid_1d(tp[2], v[2], bounds[2])
+    t_max_x = np.divide(n_x_grid - tp[0], v[0])
+    t_max_y = np.divide(n_y_grid - tp[1], v[1])
+    t_max_z = np.divide(n_z_grid - tp[2], v[2])
+    return np.array([t_max_x, t_max_y, t_max_z])
+
+def _move_to_next_voxel(idxs, tp, t_temp, t_maxs, v):
+    """
+    Move the point to next voxel.
+
+    Parameters:
+    -----------
+    idxs: list
+        List of indexes, format: [x_idx, y_idx, z_idx]
+    tp : numpy array
+        An array of size 3. Temporary point position.
+    t_temp: float
+        Temporary travel lenght from the start.
+    t_maxs: numpy array
+        The maximum travel length in current voxel for 3 directions.
+    v : numpy array
+        Direction vector from start to end.
+
+    Returns:
+    --------
+    idxs: list
+        Updated indexes.
+    tp : numpy array
+        An array of size 3. Point after moving along the direction of v.
+    t_temp : float
+        Updated travel length. Distance from the start point to updated point.
+    """
+    t_min = min(i for i in t_maxs if i > 0)
+    update_dir = list(t_maxs).index(t_min)
+    update_idx = [0, 0, 0]
+    update_idx[update_dir] = 1
+    idxs = [x + y for x, y in zip(idxs, update_idx)] 
```

Comment:
Should this always be an addition?

Reply:
Found a bug here, the `update_idx[update_dir]` could be both `1` or `-1`. This is fixed in next commit.